src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpCodeGenerator.cs

```diff
@@ -82,12 +83,22 @@ public override CodeGeneratorResult Generate()
                             csharpCodeVisitor.Accept(Tree.Children);
                         }
                     }
+
+                    BuildAfterExecuteContent(writer, Tree.Children);
                 }
             }
 
             return new CodeGeneratorResult(writer.GenerateCode(), writer.LineMappingManager.Mappings);
         }
 
+        protected virtual void BuildAfterExecuteContent(CSharpCodeWriter writer, IList<Chunk> chunks)
+        {
```

Comment:
Oh, understood. I don't think anything would blow up...  It would be similar to adding/deleting parts of the .cshtml file, maybe. Worst case, you could change the content of the chunks and rewrite the view? If the chunks were invalid, perhaps things would just not render? (I'm not sure on this, what do you think?) The chunk list has to be modifiable in our current implementation so we can modify the type names. Is this valid?